server/consumer.go

```diff
@@ -382,6 +382,16 @@ func checkConsumerCfg(
 	if config.Replicas < 0 {
 		return NewJSReplicasCountCannotBeNegativeError()
 	}
+	// If the stream is interest or workqueue retention make sure the replicas
+	// match that of the stream. This is REQUIRED for now.
+	if cfg.Retention == InterestPolicy || cfg.Retention == WorkQueuePolicy {
```

Comment:
Hi @derekcollison PR description mentioned changes only for InterestPolicy, any specific reason to consider WorkQueue here ? 
Does it mean going forward with new version we wont be able to create consumer for InterestBased as well as WorkQueue based if replica is < stream replica ?

Reply:
For now that is correct. The way the system is architected currently this is a requirement.

However, we are looking at changes for 2.11 to the complete way consumers work, so stay tuned.

Reply:
Hi @derekcollison we have a prod use case where we want to have million of consumer in total out of few hundreds workqueue stream. Our stream is R3 file and consumer is R1 memory. We cant have R3 consumer  as with millions of consumer there will be way too much of network traffic.

Also could you please share the timeline for 2.11 ?

Reply:
You can not have an R1 consumer with and R2 stream. 

We have plans to reach those goals and are working with some customers now. 

Feel free to send me an email directly and we can setup a meeting.

derek at synadia dot com